Library/Homebrew/formula_auditor.rb

```diff
@@ -865,14 +872,19 @@ def head_only?(formula)
     end
 
     def linux_only_gcc_dep?(formula)
```

Comment:
Here's an unhandled edge-case (that currently isn't used by any formula), but is probably pre-existing:
```ruby
on_system :linux, macos: :big_sur_or_newer do
  depends_on "gcc"
end
```
This is falsely flagged as a Linux-only GCC dependency.